conans/client/api/conan_api.py

```diff
@@ -90,19 +166,79 @@ def user_update(self, user_name, user_pasword):
         return {}
 
     @api_method
-    def search_recipes(self, query, remote_patterns=None, local_cache=False):
-        remote = None
-        if remote_patterns is not None and len(remote_patterns) > 0:
-            remote = remote_patterns[0].replace("*", "remote")
+    def search_local_recipes(self, query):
+        data = {
+            "remote": None,
+            "error": False,
+            "results": None
+        }
+
+        remotes = self.app.cache.registry.load_remotes()
+        search = Search(self.app.cache, self.app.remote_manager, remotes)
+
+        try:
+            references = search.search_local_recipes(query)
+        except ConanException as exc:
+            data["error"] = True
+            raise
```

Comment:
I guess it does not make sense to have the `data['error']` field because we're raising the exception anyway so we'll always have the `'error' = False` at the end of the process, won't it?

Reply:
Yes, that is true, that is probably legacy for when using multiple remotes and 1 could fail, but not the others.